specs/DisableCrashReporting.md

```diff
@@ -0,0 +1,126 @@
+Process Info
+===
+
+# Background
+Currently, when a WebView2 is created a WebView2 crashpad process will be created in addition to starting a WebView2 browser process. If any WebView2 process crashes, the crashpad process will create crash dumps and send them to Microsoft for diagnosis. This document covers new APIs to allow the end developer to customize crash reporting to help when running diagnostics and doing analysis. They can set the `CoreWebView2EnvironmentOptions.IsCustomCrashReportingEnabled` property to not have crash dumps sent to Microsoft and use the `CoreWebView2Environment.CrashDumpFolderPath` property to locate crash dumps and do customization with them instead.
+
+# Examples
+## WinRT and .NET   
+```c#
+
+/// Create WebView Environment with option
+void CreateEnvrionmentWithOption()
+{
+    CoreWebView2EnvironmentOptions options = new CoreWebView2EnvironmentOptions();
+    options.IsCustomCrashReportingEnabled = true;
+    CoreWebView2Environment environment = await CoreWebView2Environment.CreateAsync(BrowserExecutableFolder, UserDataFolder, options);
+}
+
+string _crashDumpFolder;
+
+void GetDumpFolder()
+{
+    _crashDumpFolder = webView.CoreWebView2.Environment.CrashDumpFolderPath;
+}
+
+void WebView_CoreWebView2InitializationCompleted(object sender, CoreWebView2InitializationCompletedEventArgs e)
+{
+    webView.CoreWebView2.ProcessFailed += WebView_ProcessFailed;
+}
+
+void WebView_ProcessFailed(object sender, CoreWebView2ProcessFailedEventArgs e)
+{
+    // When process failed, do custom parsing with dumps
+    ProcessNewCrashDumps(_crashDumpFolder);
+}
+
+```
+## Win32 C++
+```cpp
+void AppWindow::InitializeWebView()
+{
+    auto options = Microsoft::WRL::Make<CoreWebView2EnvironmentOptions>();
+    CHECK_FAILURE(options->put_IsCustomCrashReportingEnabled(TRUE));
+    // ... other option properties
+
+    // ... CreateCoreWebView2EnvironmentWithOptions
+}
+
+void AppWindow::ProcessFailed()
+{
+    // Get dump folder path
+    wil::com_ptr<ICoreWebView2Environment11> environment;
+    wil::unique_cotaskmem_string crashDumpFolder;
+    CHECK_FAILURE(environment->get_CrashDumpFolderPath(&crashDumpFolder));
+	
+	// Register a handler for the ProcessFailed event.
+    CHECK_FAILURE(m_webView->add_ProcessFailed(
+        Callback<ICoreWebView2ProcessFailedEventHandler>(
+            [this, &crashDumpFolder](ICoreWebView2* sender, ICoreWebView2ProcessFailedEventArgs* argsRaw)
```

Comment:
Does capturing with '&' work? Won't that get a reference to the local variable object which will go out of scope and then we'll crash in the event handler? I think you want just crashDumpFolder without the '&'?